### PR TITLE
CATROID-25 Add WriteVariableOnDeviceBrick

### PR DESCRIPTION
**Pull request by gPathpp**
New Branch because of broken GIT-history.

### CODE REVIEW

File: catroid/src/androidTest/java/org/catrobat/catroid/test/io/devicevariableaccessor/DeviceVariableAccessorCleanRoutineTest.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		sprite1.getUserVariables().addAll(generateUserVariables("S1_Variable:", 3));
+		allVariables.addAll(sprite1.getUserVariables());
+
+		sprite2.getUserVariables().addAll(generateUserVariables("S2_Variable:", 3));
+		allVariables.addAll(sprite2.getUserVariables());
+
+		sprite3.getUserVariables().addAll(generateUserVariables("S3_Variable:", 3));
+		allVariables.addAll(sprite3.getUserVariables());
+
+		project.getUserVariables().addAll(generateUserVariables("S3_Variable:", 3));
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/io/devicevariableaccessor/DeviceVariableAccessorCleanRoutineTest.java]: check name

File: catroid/src/androidTest/java/org/catrobat/catroid/test/io/devicevariableaccessor/DeviceVariableAccessorNullValueTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import java.util.HashMap;
+
+import static junit.framework.Assert.assertFalse;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+public class DeviceVariableAccessorNullValueTest {
+
+	public Object initialValue = null;
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/io/devicevariableaccessor/DeviceVariableAccessorNullValueTest.java]: declare nullUserVariable here.
kick the global/local tests below, accessor doesnt care.
remove sprite, not necessary.

File: catroid/src/androidTest/java/org/catrobat/catroid/test/io/devicevariableaccessor/DeviceVariableAccessorParameterizedValueTest.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	private UserVariable localUserVariable;
+	private Sprite sprite;
+	private DeviceVariableAccessor accessor;
+
+	@Before
+	public void setUp() {
+		directory = new File(InstrumentationRegistry.getTargetContext().getCacheDir(), "DeviceValues");
+		directory.mkdir();
+
+		sprite = new Sprite("_sprite_");
```
Comment (84n4n4) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/io/devicevariableaccessor/DeviceVariableAccessorParameterizedValueTest.java]: remove sprite and remove global/local, only test the accessor

File: catroid/src/main/java/org/catrobat/catroid/io/DeviceVariableAccessor.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+				for (UUID key: localVariableKeys) {
+					keysToRemove.remove(key);
+				}
+
+				for (UUID key: keysToRemove) {
+					deviceVariableMap.remove(key);
+				}
+
+				writeMapToJson(deviceVariableMap);
```
Comment (84n4n4) [inline catroid/src/main/java/org/catrobat/catroid/io/DeviceVariableAccessor.java]: move "try" down here.

File: catroid/src/test/java/org/catrobat/catroid/test/io/DeviceVariableAccessorExceptionTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	public void setUp() throws Exception {
+		deviceVariableAccessor = new DeviceVariableAccessor(new File("a"));
+		variableFile = Mockito.mock(File.class);
+		Mockito.when(variableFile.exists()).thenReturn(true);
+		deviceVariableAccessor.setDeviceVariablesFile(variableFile);
+		PowerMockito.whenNew(Gson.class).withAnyArguments().thenThrow(new FileNotFoundException());
+	}
+
+	@Test
+	public void deleteCorruptedFileOnReadTest() {
```
Comment (84n4n4) [inline catroid/src/test/java/org/catrobat/catroid/test/io/DeviceVariableAccessorExceptionTest.java]: please add a test that verifies the exception being thrown by writeMap

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**